bundles/org.openhab.binding.modbus.kermi/src/main/java/org/openhab/binding/modbus/kermi/internal/handler/KermiXcenterThingHandler.java

```diff
@@ -0,0 +1,666 @@
+
+    private ChannelUID powerChannel;
+    private ChannelUID powerHeatingChannel;
+    private ChannelUID powerDrinkingWaterChannel;
+    private ChannelUID powerCoolingChannel;
+
+    private ChannelUID electricPowerChannel;
+    private ChannelUID electricPowerHeatingChannel;
+    private ChannelUID electricPowerDrinkingWaterChannel;
+    private ChannelUID electricPowerCoolingChannel;
+
+    // Global State
+    // private ChannelUID globalStateChannel;
+    private ChannelUID globalStateIdChannel;
+
+    // Alarm State
+    private ChannelUID alarmStateChannel;
+
+    // Work hours
+    private ChannelUID workHoursFanChannel;
+    private ChannelUID workHoursStorageLoadingPumpChannel;
+    private ChannelUID workHoursCompressorChannel;
+
+    // PV Modulation
+    private ChannelUID pvStateChannel;
+    private ChannelUID pvPowerChannel;
+    private ChannelUID pvTargetTemperatureHeatingChannel; // READ-WRITE
+    private ChannelUID pvTargetTemperatureDrinkingWaterChannel; // READ-WRITE
+
+    private final Logger logger = LoggerFactory.getLogger(KermiXcenterThingHandler.class);
+
+    private final Parser alarmParser = new Parser(Data.DataType.ALARM_STATE);
+    private ReadStatus alarmRead = ReadStatus.NOT_RECEIVED;
+    private final Parser chargingCircuitParser = new Parser(Data.DataType.CHARGING_CIRCUIT);
+    private ReadStatus chargingCircuitRead = ReadStatus.NOT_RECEIVED;
+    private final Parser energySourceParser = new Parser(Data.DataType.ENERGY_SOURCE);
+    private ReadStatus energySourceRead = ReadStatus.NOT_RECEIVED;
+    private final Parser powerParser = new Parser(Data.DataType.POWER);
+    private ReadStatus powerRead = ReadStatus.NOT_RECEIVED;
+    private final Parser pvParser = new Parser(Data.DataType.PV);
+    private ReadStatus pvRead = ReadStatus.NOT_RECEIVED;
+    private final Parser stateParser = new Parser(Data.DataType.STATE);
+    private ReadStatus stateRead = ReadStatus.NOT_RECEIVED;
+    private final Parser workHoursParser = new Parser(Data.DataType.WORK_HOURS);
+    private ReadStatus workHoursRead = ReadStatus.NOT_RECEIVED;
+
+    private @Nullable PollTask alarmPoller;
+    private @Nullable PollTask chargingCircuitPoller;
+    private @Nullable PollTask energySourcePoller;
+    private @Nullable PollTask powerPoller;
+    private @Nullable PollTask pvPoller;
+    private @Nullable PollTask statePoller;
+    private @Nullable PollTask workHourPoller;
+
+    private List<@Nullable PollTask> pollTasks = new ArrayList<>();
+
+    private @Nullable KermiConfiguration config;
+
+    /**
+     * Communication interface to the slave endpoint we're connecting to
+     */
+    protected volatile @Nullable ModbusCommunicationInterface comms = null;
+    private int slaveId;
+
+    public KermiXcenterThingHandler(Bridge thing) {
+        super(thing);
+
+        // STATE
+        globalStateIdChannel = channelUID(thing, KermiBindingConstants.STATE_GROUP, GLOBAL_STATE_ID_CHANNEL);
```

Comment:
By using a channel list in the bindingconstants file together with a list (as member of this class) you could prevend all the class level declarations and boiler plate code here, just by looping the list from the constants 

pseudo:
```
for(String channelId in channelConstantsList) {
    String[] splitted =  channelId.split("#"); // split groups
    channelList.add(channelId, channelUID(thing, splitted[0], splitted[1]));
}